### PR TITLE
[Keyboard] Neokeys G88 hotswap & solder

### PR DESCRIPTION
**Pull request by tommyamoszhao**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tommyamoszhao** [comment]
Thank you @lesshonor , matrix info is added in info.json, modified some minor infomation mistakes.
Best, TommyZ

**Comment by tommyamoszhao** [comment]
Hi @lesshonor, sorry to bother, I built all default and via version of firmwares of both G88 solder and hotswap PCBs locally, all went well, but they still failed in QMK CI Build. May I know is there anything I still need to change?
Best, TommyZ

File: keyboards/neokeys/g88/hotswap/config.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define DIODE_DIRECTION COL2ROW
+
```
Comment (waffle87) [inline keyboards/neokeys/g88/hotswap/config.h:20]:
```suggestion
```

File: keyboards/neokeys/g88/hotswap/config.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+#define RGBLIGHT_EFFECT_RAINBOW_MOOD
+#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#define RGBLIGHT_EFFECT_SNAKE
+#define RGBLIGHT_EFFECT_KNIGHT
+#define RGBLIGHT_EFFECT_CHRISTMAS
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_RGB_TEST
+#define RGBLIGHT_EFFECT_ALTERNATING
+#define RGBLIGHT_EFFECT_TWINKLE
+#endif
```
Comment (waffle87) [inline keyboards/neokeys/g88/hotswap/config.h:74]:
```suggestion
```

File: keyboards/neokeys/g88/hotswap/info.json [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    "manufacturer": "NEO Keys",
+    "url": "www.neokeys.net",
+    "maintainer": "Quarkeys",
+    "usb": {
+        "vid": "0x4E4B",
+        "pid": "0x5058",
+        "device_version": "1.0.0"
+    },
+    "processor": "atmega32u4",
+    "bootloader": "atmel-dfu",
```
Comment (waffle87) [inline keyboards/neokeys/g88/hotswap/info.json:12]:
```suggestion
    "bootloader": "atmel-dfu",
    "diode_direction": "COL2ROW",
```

File: keyboards/neokeys/g88/readme.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+* Hardware Availability: [NEO Keys](https://www.neokeys.net/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make neokeys/g88/solder:default
+
+Flashing example for this keyboard:
+
+    make neokeys/g88/solder:default:flash
+
```
Comment (waffle87) [inline keyboards/neokeys/g88/readme.md:17]:
```suggestion

## Bootloader

```

File: keyboards/neokeys/g88/solder/config.h [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+#pragma once
+
+#define DIODE_DIRECTION COL2ROW
+
+#define MATRIX_ROW_PINS { B0, B1, B2, B3, B7, C6 }
+#define MATRIX_COL_PINS { C7, F6, F5, F4, F1, E6, D5, D2, D0, D3, D4, D1, D6, D7, B4, B5 }
+#define BACKLIGHT_PIN B6
+#define BACKLIGHT_BREATHING
+#define BACKLIGHT_LEVELS 6
+
```
Comment (waffle87) [inline keyboards/neokeys/g88/solder/config.h:26]:
```suggestion
#define MATRIX_ROW_PINS { B0, B1, B2, B3, B7, C6 }
#define MATRIX_COL_PINS { C7, F6, F5, F4, F1, E6, D5, D2, D0, D3, D4, D1, D6, D7, B4, B5 }
```

File: keyboards/neokeys/g88/solder/info.json [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    "manufacturer": "NEO Keys",
+    "url": "www.neokeys.net",
+    "maintainer": "Quarkeys",
+    "usb": {
+        "vid": "0x4E4B",
+        "pid": "0x5057",
+        "device_version": "1.0.0"
+    },
+    "processor": "atmega32u4",
+    "bootloader": "atmel-dfu",
```
Comment (waffle87) [inline keyboards/neokeys/g88/solder/info.json:12]:
```suggestion
    "bootloader": "atmel-dfu",
    "diode_direction": "COL2ROW",
    "backlight": {
        "pin": "B6",
        "levels": 6,
        "breathing": true
    },
```

File: keyboards/neokeys/g88/solder/info.json [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (waffle87) [inline keyboards/neokeys/g88/solder/info.json:14]:
```suggestion
        "LAYOUT": {
```

File: keyboards/neokeys/g88/solder/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    ),
+	[1] = LAYOUT_all(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, RGB_TOG, KC_UP,   RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, QK_BOOT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+	[2] = LAYOUT_all(
```
Comment (waffle87) [inline keyboards/neokeys/g88/solder/keymaps/via/keymap.c:36]:
```suggestion
    [0] = LAYOUT(
		KC_ESC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_F13, KC_SCRL, KC_PAUS, 
		KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_MINS, KC_EQL, KC_BSPC, KC_DEL, KC_HOME, KC_PGUP, 
		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, KC_END, KC_PGDN, 
		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_HASH, KC_ENT, KC_LEFT, KC_INS, 
		KC_LSFT, KC_BSLS, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, MO(2), KC_UP, KC_DEL, 
		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_PSCR, KC_0, KC_RALT, MO(1), KC_APP, KC_RCTL, KC_DOWN, KC_RGHT
    ),
	[1] = LAYOUT(
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
		KC_TRNS, RGB_TOG, KC_UP,   RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
		KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
		KC_TRNS, KC_TRNS, QK_BOOT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
    ),
	[2] = LAYOUT(
```

File: keyboards/neokeys/g88/solder/rules.mk [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+RGBLIGHT_ENABLE = yes		# Enable RGB Backlight
```
Comment (waffle87) [inline keyboards/neokeys/g88/solder/rules.mk:13]:
```suggestion
```

File: keyboards/neokeys/g88/solder/solder.h [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include "quantum.h"
+#define XXXX KC_NO
+
+#define LAYOUT_all( \
```
Comment (waffle87) [inline keyboards/neokeys/g88/solder/solder.h:21]:
```suggestion
#define LAYOUT( \
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by tommyamoszhao** [comment]
Hi @drashna, matrix_pins info has been moved into info.json, .h file is deleted and copyright info is corrected. Thx TommyZ

File: keyboards/neokeys/g88/hotswap/config.h [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define MATRIX_ROW_PINS { B0, B1, B2, B3, B7, C6 }
+#define MATRIX_COL_PINS { C7, F6, F5, F4, F1, E6, D5, D2, D0, D3, D4, D1, D6, D7, B4, B5 }
```
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/neokeys/g88/hotswap/config.h:20]:
```suggestion
```

**Comment by tommyamoszhao** [comment]
hi @drashna , both .h files are deleted and matrix info is moved into info.json. :)

**Pull request closed without merging**